x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResponseListener.java

```diff
@@ -164,6 +166,7 @@ public ActionListener<EsqlQueryResponse> wrapWithLogging() {
             long timeMillis = stopWatch.stop().getMillis();
             onFailure(ex);
             LOGGER.info("Failed execution of ESQL query.\nQuery string: [{}]\nExecution time: [{}]ms", esqlQuery, timeMillis);
+            logQueryFailure(LOGGER, LOGGING_PREFIX, ex, null);
```

Comment:
I have no idea why I didn't log before sending out the failure response :facepalm: but I think we can do the same here.

Reply:
Updated this too.